src/factory/createStorageHook.ts

```diff
@@ -31,25 +31,33 @@ const createStorageHook = (type: 'session' | 'local') => {
       }
       return [JSON.stringify(defaultValue) as unknown as TValue, noop]
     }
+    const storage = (window)[storageName]
+
+    const safelySetStorage = (valueToStore: string) => {
+      try {
+        storage.setItem(storageKey, valueToStore)
+        // eslint-disable-next-line no-empty
+      } catch (e) {}
+    }
 
-    const storage = (window as any)[storageName]
     const [storedValue, setStoredValue] = useState<TValue>(
       () => {
+        let valueToStore: string
         try {
-          return safelyParseJson(storage.getItem(storageKey) || JSON.stringify(defaultValue))
```

Comment:
`storage.getItem` will return `null` if the key does not exist, which is more narrowly checked with `??` vs `||`.  for example,`||` would be falsey if the key _did_ exist and the value was an empty string. Although an empty string may be a legitimate saved value, this code would overwrite it with the `defaultValue` instead.